src/websocket-client.ts

```diff
@@ -997,6 +1069,52 @@ export class WebsocketClient extends EventEmitter {
     }
   }
 
+  private updatePendingTopicSubscriptionStatus(wsKey: string, msg: any) {
+    const requestsIds = msg['req_id'] as string;
+    const pendingTopicsSubscriptions = this.pendingTopicsSubscriptions.find(
+      (s) => s.wsKey === wsKey,
+    );
+    if (!pendingTopicsSubscriptions) {
+      throw new Error(
+        `Could not find "${wsKey}" within pending topics subscriptions.`,
+      );
+    }
+
```

Comment:
It's important that any logic in the message handler doesn't prevent the message from reaching the consumer. That also means preventing exceptions from blocking that workflow. 

In this case, if an event comes in (position update, for example) and for some weird reason there's no pending subscriptions (maybe because it's a success already), this would throw and prevent any further events on that ws key (missed position updates).

I think it might be good to wrap this whole fn in a try catch, so even if there is an exception. it's a non-destructive one that's logged but doesn't block any existing workflows. Would also be curious to test this to be sure it doesn't introduce any excessive logging. E.g. when might pending topics be missing?

Reply:
I was in doubt if throwing an error there, or just log what happened and let the caller handle it, but also, that scenario should never happen. Actually, the original error message was ending with a "this should not happen". Therefore if the caller uses the `subscribe` or `subscribeV5` there should be no problems. Since before calling that method the code handling the messages coming from the exchange first checks that the message event is actually a topic sub confirm by calling first `isTopicSubscriptionConfirmation`.

My worring was about a potential future change in the implementation where we could forget to update the pending subscriptions list, and then the event coming from the exchange would have thrown that error, blowing everything up. I've taken for granted that the publishing process would have built and ran the tests on the code. And when reaching the spec for the ws subcscription it would have failed. 

But yeah, it could be a little overkilling and paranoid. I could completely remove that second check, or just log it if we want to keep that second check?

So to reply at "E.g. when might pending topics be missing?"

When we or whoever touches the codebase forgets to add that pending topic subscription. The final caller could never get this wrong.